lib/Common/DataStructures/BaseDictionary.h

```diff
@@ -860,9 +860,14 @@ namespace JsUtil
             int initSize = max(capacity, 4);
             uint initBucketCount = SizePolicy::GetBucketSize(initSize);
             AssertMsg(initBucketCount > 0, "Size returned by policy should be greater than 0");
-            Allocate(&buckets, &entries, initBucketCount, initSize);
 
-            // Allocation can throw - assign the size only after allocation has succeeded.
+            int* newBuckets = nullptr;
+            EntryType* newEntries = nullptr;
+            Allocate(&newBuckets, &newEntries, initBucketCount, initSize);
+
+            // Allocation can throw - assign only after allocation has succeeded.
+            SetArray<TAllocator>(this->buckets, newBuckets, initBucketCount);
```

Comment:
Yes, `this->entries = newEntries` is fine when `newEntries` already takes care of write barrier on the array elements. So either I do it here with `SetArray`, or I do something similar ahead replacing `js_memcpy_s` with a version of `CopyArray`. Either approach has the concern that it is hard to know we missed one. Do you think `CopyArray` is safer in terms of write barrier correctness? e.g. is it possible GC collects in between `js_memcpy_s` and my current `SetArray`, and `SetArray` write barrier is too late?